File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end()); 
        string answer = "";
        int m = strs.size()-1;  
        for(int i = 0; i < strs[0].length(); i++){
            if(strs[0][i] == strs[m][i]) answer += strs[0][i]; 
            else break; 
        }
        return answer; 
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs); 
        StringBuilder sb = new StringBuilder(strs[0]);
        int len = sb.length(); 
        //System.out.print(len); 
        for(int i = 1; i < strs.length; i++){
            while(!strs[i].startsWith(sb.toString())){
                sb = new StringBuilder(sb.substring(0,len--)); 
                if(len < 0) return ""; 
            }
        }
        return sb.toString(); 
    }
}